src/communication/WebSocketSession.ts

```diff
@@ -9,30 +9,22 @@ import { Session } from "../interfaces/Session";
 import * as uuid from "uuid/v4";
 
 export class WebSocketSession implements Session {
-    private sessionData: string;
     private publishers: Array<WebSocketPublisher>;
     private subscribers: Array<WebSocketSubscriber>;
     private privateTopics: Array<string>;
     private configuration: ApiConfiguration;
     private stateMachineRefSendPublisher: WebSocketPublisher;
-    public heartbeatIntervalSeconds: number;
     public privateTopic: string;
-    public closedByUser: boolean;
-    public heartbeatTimer: number;
     public privateSubscriber: Subscriber;
-    public webSocket: WebSocket;
 
-    constructor(webSocket: WebSocket, sessionData?: string) {
-        this.webSocket = webSocket;
+    constructor(public webSocket: WebSocket, private sessionData?: string) {
```

Comment:
I don't get this mix of static non-static... what's the underlying problem?

Reply:
I don't get why different connections should share information. (Because they share the same instance of WebSocketSection static object)